server/middleware/namespace.go

```diff
@@ -18,55 +18,54 @@ import (
-	api "github.com/tigrisdata/tigris/api/server/v1"
 	"github.com/tigrisdata/tigris/lib/container"
-	"github.com/tigrisdata/tigris/server/config"
-	"github.com/tigrisdata/tigris/server/metadata"
 	"github.com/tigrisdata/tigris/server/request"
 	"google.golang.org/grpc"
 )
 
-type NamespaceSetter struct {
-	tenantManager      *metadata.TenantManager
-	namespaceExtractor request.NamespaceExtractor
-	excludedMethods    container.HashSet
-	config             *config.Config
-}
+var (
+	excludedMethods = container.NewHashSet(
+		"/HealthAPI/Health",
+		"/tigrisdata.admin.v1.Admin/createNamespace",
+		"/tigrisdata.admin.v1.Admin/listNamespaces",
+	)
+
+	namespaceExtractor = &request.AccessTokenNamespaceExtractor{}
+)
 
-func (r *NamespaceSetter) NamespaceSetterUnaryServerInterceptor() func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+func namespaceSetterUnaryServerInterceptor(enabled bool) func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-		if !r.config.Auth.EnableNamespaceIsolation || r.excludedMethods.Contains(info.FullMethod) {
-			return handler(request.SetNamespace(ctx, request.DefaultNamespaceName), req)
-		} else {
-			namespace, err := r.namespaceExtractor.Extract(ctx)
-			if err != nil {
-				return nil, err
-			}
-			if namespace == "" {
-				return handler(request.SetNamespace(ctx, "unknown"), req)
+		namespace := request.DefaultNamespaceName
+		if enabled && !excludedMethods.Contains(info.FullMethod) {
+			var err error
+			if namespace, err = namespaceExtractor.Extract(ctx); err != nil {
+				// We know that getAccessToken with app_id/app_secret credentials doesn't have namespace set.
+				// Mark it as default_namespace instead of unknown.
+				if info.FullMethod != "/tigrisdata.auth.v1.Auth/getAccessToken" {
+					// We return and error when the token is set, but namespace is empty
+					return nil, err
+				}
+				namespace = request.DefaultNamespaceName
 			}
-			return handler(request.SetNamespace(ctx, namespace), req)
 		}
+
+		return handler(request.SetNamespace(ctx, namespace), req)
 	}
 }
 
-func (r *NamespaceSetter) NamespaceSetterStreamServerInterceptor() grpc.StreamServerInterceptor {
+func namespaceSetterStreamServerInterceptor(enabled bool) grpc.StreamServerInterceptor {
 	return func(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
-		if !r.config.Auth.EnableNamespaceIsolation {
-			wrapped := middleware.WrapServerStream(stream)
-			wrapped.WrappedContext = request.SetNamespace(stream.Context(), request.DefaultNamespaceName)
-			return handler(srv, wrapped)
-		} else {
-			namespace, err := r.namespaceExtractor.Extract(stream.Context())
-			if err != nil {
+		namespace := request.DefaultNamespaceName
```

Comment:
does this mean if no namespace is set it would default to `default_namespace`?

Reply:
This is intended behavior:
* Return and error if namespace is empty in the token
* Set namespace to "unknown" if token is not present in the request
* Set namespace to "default_namespace" if we know that token should not
be present or request is across namespaces

Reply:
First two handled by auth, it doesn't matter much at this level.

Reply:
I don't think we should use "default_namespace" for it. This will be confusing because now we are using default_namespace for internal metadata, the default for the users where they don't want isolation, and across the namespaces now. Probably OK for now but let's think about it.